Mathlib/LinearAlgebra/Dimension/Basic.lean

```diff
@@ -379,6 +379,29 @@ lemma rank_le_of_isSMulRegular {S : Type*} [CommSemiring S] [Algebra S R] [Modul
 @[deprecated (since := "2024-11-21")]
 alias rank_le_of_smul_regular := rank_le_of_isSMulRegular
 
+lemma Module.rank_top_le_rank_of_isScalarTower
```

Comment:
I think you can extract R and S and their instances to `variable`s. I also think you can remove Comm and replace Algebra by Module (maybe you'll need IsScalarTower R S S though).